jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -148,20 +141,90 @@ private static boolean isProgressFooterEnabled(Project project) {
     this.project = project;
-    this.javaLayerConfigurations = javaLayerConfigurations;
+    this.logger = logger;
+    this.appRoot = appRoot;
 
     eventHandlers = makeEventHandlers(project, logger, singleThreadedExecutor);
   }
 
   @Override
-  public JavaLayerConfigurations getJavaLayerConfigurations() {
-    return javaLayerConfigurations;
+  public JibContainerBuilder getContainerBuilderWithLayers(RegistryImage baseImage) {
+    try {
+      if (isWarProject()) {
+        logger.info("WAR project identified, creating WAR image: " + project.getDisplayName());
+        Path explodedWarPath = GradleProjectProperties.getExplodedWarDirectory(project);
+        return JavaContainerBuilderHelper.fromExplodedWar(baseImage, explodedWarPath, appRoot);
+      }
+
+      JavaPluginConvention javaPluginConvention =
+          project.getConvention().getPlugin(JavaPluginConvention.class);
+      SourceSet mainSourceSet =
+          javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
+
+      FileCollection classesOutputDirectories =
+          mainSourceSet.getOutput().getClassesDirs().filter(File::exists);
+      Path resourcesOutputDirectory = mainSourceSet.getOutput().getResourcesDir().toPath();
+      FileCollection allFiles = mainSourceSet.getRuntimeClasspath();
+      FileCollection dependencyFiles =
+          allFiles
+              .minus(classesOutputDirectories)
+              .filter(file -> !file.toPath().equals(resourcesOutputDirectory));
+
+      JavaContainerBuilder javaContainerBuilder =
+          JavaContainerBuilder.from(baseImage).setAppRoot(appRoot);
+
+      // Adds resource files
+      if (Files.exists(resourcesOutputDirectory)) {
+        javaContainerBuilder.addResources(resourcesOutputDirectory);
+      }
+
+      // Adds class files
+      for (File classesOutputDirectory : classesOutputDirectories) {
+        javaContainerBuilder.addClasses(classesOutputDirectory.toPath());
+      }
+      if (classesOutputDirectories.isEmpty()) {
+        logger.warn("No classes files were found - did you compile your project?");
+      }
+
+      // Adds dependency files
+      javaContainerBuilder.addDependencies(
+          dependencyFiles
+              .getFiles()
+              .stream()
+              .filter(File::exists)
+              .map(File::toPath)
+              .collect(Collectors.toList()));
+
+      return javaContainerBuilder.toContainerBuilder();
+
+    } catch (IOException ex) {
+      throw new GradleException("Obtaining project build output files failed", ex);
+    }
+  }
+
+  @Override
+  public List<Path> getClassFiles() throws IOException {
+    // TODO: Consolidate with getContainerBuilderWithLayers
+    JavaPluginConvention javaPluginConvention =
+        project.getConvention().getPlugin(JavaPluginConvention.class);
+    SourceSet mainSourceSet = javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
```

Comment:
I'm pretty ignorant of sourceSets — we don't walk any other sourceSets?

Reply:
Nope, I think this matches previous behavior.